### PR TITLE
Tutorial for rendered doc regeneration

### PR DESCRIPTION
**Pull request by AlexandreF-1qbit**
Added a `dev_tools` folder that contains a tutorial file for using `quarto`.

### CODE REVIEW

File: dev_tools/generate_documentation.rst
Comment (ValentinS4t1qbit) [inline dev_tools/generate_documentation.rst:4]: project maintainers

File: dev_tools/generate_documentation.rst
Comment (ValentinS4t1qbit) [inline dev_tools/generate_documentation.rst:6]: We describe below how to re-generate this documentation after changes in the repository, such as the modification or addition of new notebooks.

File: dev_tools/generate_documentation.rst
Comment (ValentinS4t1qbit) [inline dev_tools/generate_documentation.rst:32]: `README.rst` : A README update for users visiting the Github page.

File: dev_tools/generate_documentation.rst
Comment (ValentinS4t1qbit) [inline dev_tools/generate_documentation.rst:33]: `index.qmd` : the main page of the rendered documentation, listing and tying together all the other rendered html files.

File: dev_tools/generate_documentation.rst
Comment (ValentinS4t1qbit) [inline dev_tools/generate_documentation.rst:38]: Skip a line and add:

Note that all html pages include data about the side menu used to navigate through the website. As a result, even though you may be only adding or modifying a single file, you may find that all generated files (html, json) have slightly changed as a result. Please make sure to add, commit and push all modifications, and trust the process.

File: dev_tools/generate_documentation.rst
Comment (ValentinS4t1qbit) [inline dev_tools/generate_documentation.rst:47]: At the time of writing this tutorial, only way to regenerate the documentation is to manually deploy from the user's branch. Regenerating this documentation and verifying it works as expected is mandatory before approving and merging a PR that impacts it. Doing so will however overwrite the existing documentation. In case the new documentation is not working out and needs further work, please fix it as soon as possible or regenerate from the `main` branch to revert to the previous version: many users depend on it !

**Review by ValentinS4t1qbit — CHANGES_REQUESTED** [review]
Thank you, here are some suggestions for the text. You did a great job, if we manage to understand why this adapt test is failing we may be golden.

I recall @JamesB-1qbit made some changes with how keywords are parsed in our solvers: maybe we accidentally stopped supporting "spin" or whatever is used here, or maybe we need to update the code in the notebook to call it adequately.

File: dev_tools/generate_documentation.rst
Comment (JamesB-1qbit) [inline dev_tools/generate_documentation.rst:51]: ... this tutorial, only way ... -> ... this tutorial, the only way ...

**Review by JamesB-1qbit — CHANGES_REQUESTED** [review]

**Review by JamesB-1qbit — APPROVED** [review]

File: dev_tools/generate_documentation.rst
Comment (ValentinS4t1qbit) [inline dev_tools/generate_documentation.rst:33]: The

File: dev_tools/generate_documentation.rst
Comment (ValentinS4t1qbit) [inline dev_tools/generate_documentation.rst:1]: Render the online documentation with Quarto

**Review by ValentinS4t1qbit — APPROVED** [review]

**Pull request merged**
